ui/src/app/core/capabilities-activator.service.ts

```diff
@@ -0,0 +1,72 @@
+import {ActivatedRouteSnapshot, RouterStateSnapshot, Router} from '@angular/router';
+import {CanActivate} from '@angular/router';
+import {Injectable} from '@angular/core';
+
+import {AuthService} from './auth.service';
+import {CapabilitiesResponse} from '../shared/model/CapabilitiesResponse';
+import {CapabilitiesService} from './capabilities.service';
+import {environment} from '../../environments/environment';
+import {URLSearchParamsUtils} from '../shared/utils/url-search-params.utils'
+
+/** Service wrapper for routing based on current authentication state. */
+@Injectable()
+export class CapabilitiesActivator implements CanActivate {
+
+  constructor(
+    private readonly authService: AuthService,
+    private readonly capabilitiesService: CapabilitiesService,
+    private readonly router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<boolean> {
+    return this.capabilitiesService.getCapabilities()
+      .then(capabilities =>
+        this.handleAuthCapabilities(
+          capabilities, route.routeConfig.path, state.url))
+      .then(result => result[0] == false
```

Comment:
`.then([authenticated, capabilities] => ...`

Reply:
Not supported in typscript unfortunately :(